app/services/checksum_validator.rb

```diff
@@ -2,86 +2,81 @@
 
 # code for validating Moab checksums
 class ChecksumValidator
-  attr_reader :bare_druid, :results, :complete_moab
+  attr_reader :druid, :results, :complete_moab
 
-  alias druid bare_druid
   delegate :moab_storage_root, to: :complete_moab
   delegate :storage_location, to: :moab_storage_root
-  delegate :mark_moab_not_found, :moab, :object_dir, :set_status_as_seen_on_disk, :update_status, to: :moab_validator
-
-  MANIFESTS = 'manifests'
-  MANIFESTS_XML = 'manifestInventory.xml'
-  MODIFIED = 'modified'
-  ADDED = 'added'
-  DELETED = 'deleted'
-  FILES = 'files'
+  delegate :moab, :object_dir, to: :moab_validator
 
   def initialize(complete_moab)
     @complete_moab = complete_moab
-    @bare_druid = complete_moab.preserved_object.druid
-    @results = AuditResults.new(bare_druid, nil, moab_storage_root, 'validate_checksums', logger: Audit::Checksum.logger)
+    @druid = complete_moab.preserved_object.druid
+    @results = AuditResults.new(druid, nil, moab_storage_root, 'validate_checksums', logger: Audit::Checksum.logger)
     # TODO: fix fragile interdependence, MoabValidator wants AuditResults instance, but we want MoabValidator#moab.current_version_id
     # in that AuditResults instance.  so set AuditResults#actual_version after both instances have been created.
     @results.actual_version = moab.current_version_id
   end
 
   def validate_checksums
     # check first thing to make sure the moab is present on disk, otherwise weird errors later
-    return results.result_array unless check_file_existence!
+    return persist! { moab_validator.mark_moab_not_found } unless moab_exists?
 
+    # These will populate the results object
     validate_manifest_inventories
     validate_signature_catalog
 
-    # This is to deal with db connection timeouts.
-    ActiveRecord::Base.clear_active_connections!
-
-    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+    persist!(clear_connections: true) do
```

Comment:
- "persist!" as a method concept is great;  the notion that it's about database transaction is less obvious, tho.  Maybe that's ok?  Or maybe `persist_db_transaction!` ?   I'm open to other opinions.

Reply:
Changed to `persist_db_transaction!`.